libsplinter/src/admin/service/mod.rs

```diff
@@ -131,10 +131,11 @@ impl AdminKeyVerifier for dyn RegistryReader {
     /// registry and the node has the given key. Otherwise, the key is not permitted.
     fn is_permitted(&self, node_id: &str, key: &[u8]) -> Result<bool, AdminKeyVerifierError> {
         let node_opt = self.fetch_node(node_id).map_err(|err| {
-            AdminKeyVerifierError::new_with_source(
-                &format!("Failed to lookup node '{}' in registry", node_id),
-                Box::new(err),
-            )
+            AdminKeyVerifierError::new(&format!(
```

Comment:
Wondering if we can use a pattern that doesn't require this error to be Send, allowing it to keep the new_with_source() method. Maybe handling it at the channel boundary by not sending AdminKeyVerifierError directly, but sending its display and debug instead.

Reply:
We should have a larger discussion about this pattern and update this in another PR based on that conversation.

Reply:
I think we avoid having to do this in some endpoints by not using `web::block` and instead just converting the HTTP response into a future like `return Box::new(HttpResponse::Ok().finish().into_future())`